### PR TITLE
gpt4all: 2.8.0 -> 3.2.1

### LINKED ISSUE #335993: Update request: gpt4all 2.8.0 → 3.2.1
- Package name: gpt4all
- Latest released version: 3.2.1

- Current version on the unstable channel: 2.8.0
- Current version on the stable/release channel: 2.7.5

- [ X ] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@drupol 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Titaniumtown**
## Description of changes
[Changelog](https://github.com/nomic-ai/gpt4all/compare/v2.8.0...v3.2.1)
closes: #335993

## Things done
Updated gpt4all and added a patch to disable compile-time fetching of an embeddings model, shifted that to be done directly in the nix package.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Titaniumtown** [comment]
I've tested functionality locally on nixOS w/o any issues!

**Review by ryand56 — COMMENTED** [review]
PR title should be: `gpt4all: 2.8.0 -> 3.2.1`

**Comment by Titaniumtown** [comment]
@ryand56 fixed. lol. ty for pointing that out :)

**Comment by Titaniumtown** [comment]
@ryand56 does it all look good to you now?

**Comment by ryand56** [comment]
Looks good :) I'll test it later

**Comment by reedriley** [comment]
When I apply this patch and build on x86_64-linux, I end up with a file named `60c2zplz1cxvsy27xg7sr5nxirqpkmlr-nomic-embed-text-v1.5.f16.gguf` (with a prepended Nix store hash) in `$out/resources/` - which the resulting binary won't look for.

**Comment by reedriley** [comment]
Here's what ended up working for me, in case it helps: https://github.com/NixOS/nixpkgs/commit/e56a0c7308651532fc2b666d56545b89a01b4bb2

**Comment by Titaniumtown** [comment]
@reedriley thanks for the feedback! I'll apply your changes tomorrow. Good catch!

**Comment by Titaniumtown** [comment]
@reedriley just pushed your changes, ty <3

File: pkgs/by-name/gp/gpt4all/package.nix [hunk 1]
```diff
@@ -56,6 +67,7 @@ stdenv.mkDerivation (finalAttrs: {
     "-DKOMPUTE_OPT_USE_BUILT_IN_VULKAN_HEADER=OFF"
     "-DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK=ON"
     "-DKOMPUTE_OPT_USE_BUILT_IN_FMT=OFF"
+    "-DLOCAL_EMBEDDING_MODEL_PATH=${finalAttrs.embed_model}"
```
Comment (reedriley) [inline pkgs/by-name/gp/gpt4all/package.nix:70]: This line is probably unnecessary with the CMakeLists.txt changes.

Comment (Titaniumtown) [reply pkgs/by-name/gp/gpt4all/package.nix]: Removed.

**Comment by reedriley** [comment]
LGTM!

**Review by ryand56 — APPROVED** [review]

**Comment by Titaniumtown** [comment]
@polygon any thoughts? the bot added you to review this :p

**Comment by polygon** [comment]
> @polygon any thoughts? the bot added you to review this :p

Looks like @drupol pulled a nice bait and switch on me. I maintained this out-of-tree because I only sporadically find time for maintenance work. Then he pushed for adoption into Nixpkgs and removed himself as a maintainer shortly after.

I will look into this, but it can take some time. If anyone wants to co-maintain this package, feel free to take over.

**Comment by drupol** [comment]
@polygon Sorry mate, it's not intentional at all... I try to avoid maintaining stuff I don't use, or else it's too much work for me.

Regarding this GPT4All, I used to use it until I switched to `open-webui` that I maintain actively.

**Comment by Titaniumtown** [comment]
> > @polygon any thoughts? the bot added you to review this :p
> 
> Looks like @drupol pulled a nice bait and switch on me. I maintained this out-of-tree because I only sporadically find time for maintenance work. Then he pushed for adoption into Nixpkgs and removed himself as a maintainer shortly after.
> 
> I will look into this, but it can take some time. If anyone wants to co-maintain this package, feel free to take over.

I can work on it more if needed :) I want some more nix experience.

**Review by felschr — CHANGES_REQUESTED** [review]
Commit message should be:
```diff
-gpt4all 2.8.0 -> 3.2.1
+gpt4all: 2.8.0 -> 3.2.1
```

@Titaniumtown if you wanna help maintain this package, you can also add yourself to the [`maitnainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) & `meta.maintainers` of this package.

**Comment by Titaniumtown** [comment]
@felschr I've added myself now! Thank you :)

**Comment by felschr** [comment]
@Titaniumtown adding yourself to the maintainers list & to `meta.maintainers` should be in separate commits, i.e. there should be 3 commits in total. Some of this is explained in [`CONTRIBUTING.md`](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) & [`maintainers/README.md`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) but you can also check the git history for how people did this before.

**Comment by Titaniumtown** [comment]
oh so sorry. I'll fix that right now.

**Comment by felschr** [comment]
Also as you can see in the checks, `maintainer-list.nix` needs to be formatted.

**Review by felschr — APPROVED** [review]

**Comment by ToasterUwU** [comment]
Result of `nixpkgs-review pr 337747` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gpt4all</li>
    <li>gpt4all-cuda</li>
  </ul>
</details>

**Review by ToasterUwU — APPROVED** [review]

**Comment by felschr** [comment]
@Titaniumtown thank you for the update and welcome as a package maintainer 🥳!

**Comment by Titaniumtown** [comment]
Thank you so much @felschr! I'm excited!

**Pull request merged**